src/silx/resources/__init__.py

```diff
@@ -233,18 +215,27 @@ def resource_filename(resource):
     example "silx:foo.png" identify the resource "foo.png" from the resource
     directory "silx". See also :func:`register_resource_directory`.
 
-    :param str resource: Resource path relative to resource directory
-                         using '/' path separator. It can be either a file or
-                         a directory.
+    :param resource: Resource path relative to resource directory
+                     using '/' path separator. It can be either a file or
+                     a directory.
     :raises ValueError: If the resource name uses an unregistred resource
         directory name
     :return: Absolute resource path in the file system
-    :rtype: str
     """
     return _resource_filename(resource, default_directory=None)
 
 
-def _resource_filename(resource, default_directory=None):
+# Manage resource files life-cycle
+_file_manager = contextlib.ExitStack()
+atexit.register(_file_manager.close)
```

Comment:
See https://importlib-resources.readthedocs.io/en/latest/migration.html#pkg-resources-resource-filename for explanation